lib/net.js

```diff
@@ -1018,21 +1020,25 @@ function lookupAndConnect(self, options) {
   var localPort = options.localPort;
 
   if (localAddress && !cares.isIP(localAddress)) {
-    throw new TypeError('"localAddress" option must be a valid IP: ' +
-                        localAddress);
+    throw new errors.Error('ERR_INVALID_IP_ADDRESS', localAddress);
   }
 
   if (localPort && typeof localPort !== 'number') {
-    throw new TypeError('"localPort" option should be a number: ' + localPort);
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                               'options.localPort',
+                               'number',
+                               localPort);
   }
 
   if (typeof port !== 'undefined') {
     if (typeof port !== 'number' && typeof port !== 'string') {
-      throw new TypeError('"port" option should be a number or string: ' +
-                          port);
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                                 'options.port',
+                                 ['number', 'string'],
+                                 port);
     }
     if (!isLegalPort(port)) {
-      throw new RangeError('"port" option should be >= 0 and < 65536: ' + port);
+      throw new errors.RangeError('ERR_SOCKET_BAD_PORT');
```

Comment:
Should the `port` be passed as a parameter to the `throw` here, and also on line 1475?

Reply:
I think that would be nice :+1:

Reply:
The 'ERR_SOCKET_BAD_PORT' definition in `lib/internal/errors.js`
does not handle a port argument. Should I refactor it to something like this:
```E('ERR_SOCKET_BAD_PORT', (port) => {
  const portMsg = port ? `: ${port}` : '';
  return `Port${portMsg} should be > 0 and < 65536`;
});
```

Reply:
I commit the change to pass the port value in the `'ERR_SOCKET_BAD_PORT'` call in `lib/net.js`. This will not be included in the err msg until we modify the `E('ERR_SOCKET_BAD_PORT', 'Port should be > 0 and < 65536');` in `lib/internal/errors.js`(see my above comment/question)